cli/flox/src/commands/environment.rs

```diff
@@ -1602,14 +1602,58 @@ impl Upgrade {
 
 #[derive(Bpaf, Clone, Debug)]
 pub struct Containerize {
-    #[allow(unused)]
     #[bpaf(external(environment_select), fallback(Default::default()))]
     environment: EnvironmentSelect,
+
+    /// Path to write the container to (pass '-' to write to stdout)
+    #[bpaf(short, long, argument("path"))]
+    output: Option<PathBuf>,
 }
 impl Containerize {
-    pub async fn handle(self, _flox: Flox) -> Result<()> {
+    pub async fn handle(self, flox: Flox) -> Result<()> {
         subcommand_metric!("containerize");
 
-        todo!("this command is planned for a future release");
+        let mut env = self
+            .environment
+            .detect_concrete_environment(&flox, "upgrade")?
+            .into_dyn_environment();
+
+        let output_path = match self.output {
+            Some(output) => output,
+            None => std::env::current_dir()
+                .context("Could not get current directory")?
+                .join(format!("{}-container.tar.gz", env.name())),
+        };
+
+        let output: Box<dyn Write> = if output_path == Path::new("-") {
+            debug!("writing container to stdout");
+
+            Box::new(std::io::stdout())
+        } else {
+            debug!("writing container to {}", output_path.display());
+
+            let file = fs::OpenOptions::new()
+                .write(true)
+                .create(true)
+                .truncate(true)
+                .open(&output_path)
+                .context("Could not open output file")?;
+
+            Box::new(file)
+        };
+
+        let builder = Dialog {
+            message: &format!("Building container for environment {}...", env.name()),
+            help_message: None,
+            typed: Spinner::new(|| env.build_container(&flox)),
+        }
+        .spin()
+        .context("could not create container builder")?;
+
+        builder
+            .stream_container(output)
```

Comment:
note nonblocking: we should abort if stdout closes because the process we're piping to fails